packages/design-system/src/components/ChoiceList/Choice.jsx

```diff
@@ -59,11 +67,11 @@ export class Choice extends React.PureComponent {
 
     if (!this.isControlled) {
       this.setState({ checked: evt.target.checked });
-    }
 
-    if (this.uncheckEventName && evt.target.checked) {
-      // Emit the uncheck event so other radio options update their state
-      dsChoiceEmitter.emitEvent(this.uncheckEventName, [this.id]);
+      if (this.uncheckEventName && evt.target.checked) {
```

Comment:
I wonder if this event emitter logic could be more clear. It seems like `this.uncheckEventName` is always the same, and can pretty much be replaced by `${this.props.name}-uncheck`. If we did that, we can remove `this.uncheckEventName` from this if statement and replace it with `this.props.type === 'radio'`. This might make the code more clear that all this event emitter logic only happens for uncontrolled radio buttons

Reply:
We also don't have to get rid of `this.uncheckEventName`, i'd just prefer not to use it in this if statement

Reply:
Great suggestion.